openedx/features/course_experience/urls.py

```diff
@@ -63,4 +64,9 @@
         dismiss_welcome_message,
         name='openedx.course_experience.dismiss_welcome_message',
     ),
+    url(
+        r'^set_course_goal$',
```

Comment:
Here's the start to feedback on the API.  You can try to get started on this and/or maybe first get some answers to open questions from @andy-armstrong, and if not, maybe in architecture or dev channels.

1. Check out the [API conventions](https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/18350757/edX+REST+API+Conventions).
2. Implement using django rest framework.
3. See example for [bookmarks api](https://github.com/edx/edx-platform/blob/93235d118d1c86398478d4e8485a924613f5ff34/openedx/core/djangoapps/bookmarks/views.py#L124-L129).
4. Conventions use singular in the API name (like discussion_api).

**Open Questions:**

- Do we also need an api.py?  @andy-armstrong?  I feel like I've heard different things around this.
- Is it ok for this api to live in lms/course_goals with the rest of the functionality?

See comments below for potential REST-ish API calls:
```
# Not needed yet so don't implement.
# Could list all goals course goals or filter by course.
GET /api/coursegoal/v0/coursegoals/?course_id={course_id}

# Sets a user's course goal.  Assumes logged in user.
POST /api/coursegoal/v0/coursegoals/
Request data: {"course_id": <course-id>, "goal": <goal>}
```